conda/core/solve.py

```diff
@@ -268,6 +279,73 @@ def solve_final_state(self, update_modifier=NULL, deps_modifier=NULL, prune=NULL
 
         return ssc.solution_precs
 
+    def determine_constricting_specs(self, spec, solution_precs):
+        highest_version = [VersionOrder(sp.version) for sp in solution_precs
+                           if sp.name == spec.name][0]
+        constricting = []
+        for prec in solution_precs:
```

Comment:
I think I could have made use of `_get_package_pool` here but would have needed to loop through everything again to split out specs with compound version specs. I figure going through once is probably better then doing so twice